app/extensions/safe/components/authRequest.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react';
+// TODO: When I import the following, an error is thrown in CodeMirror, used by nessie-ui, that navigator is not defined
+//import { Text, Column } from 'nessie-ui';
+
+export const createAuthRequestElement = (authReqData) =>
+{
+  const R = React.createElement;
+  let reqType = 'authReq';
+  let reqTypeText = ' requests Auth Permission';
+  if (authReqData.contReq)
+  {
+    reqType = 'contReq'; 
+    reqTypeText = ' requests Container Access';
+  }
+
+  if (authReqData.mDataReq)
+  {
+    reqType = 'mDataReq'; 
+    reqTypeText = ' requests mData Access';
+  }
+
+  let children =
+  [
+    R('p', { className: 'blockText', key: authReqData[reqType].app.name }, `${authReqData[reqType].app.name} ${reqTypeText}`),
```

Comment:
@hunterlester let's use JSX here instead of react.createClass everywhere so it's consistent.